portfolio/src/main/webapp/photo-gallery.js

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+let slideIndex = 1;
+showSlides(slideIndex);
+
+// Next/previous controls called in html onclick
+function plusSlides(n) {
+  showSlides(slideIndex += n);
+}
+
+function showSlides(n) {
+  let i;
+  let slides = document.getElementsByClassName("myPhotos");
+  let markers = document.getElementsByClassName("marker");
+
+  //go to first picture at end of slide deck
+  if (n > slides.length) {slideIndex = 1}
+
+  //got to last picture from the start of deck
+  if (n < 1) {slideIndex = slides.length}
+
+  //ensure no pictures are displayed
+  for (i = 0; i < slides.length; i++) {
```

Comment:
Can you use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map?

Reply:
Because slides and markers are objects I can't use map but I can use forEach instead?